components/bio-formats-tools/src/loci/formats/tools/ImageConverter.java

```diff
@@ -591,6 +591,21 @@ public boolean testConvert(IFormatWriter writer, String[] args)
       height = Math.min(reader.getSizeY(), height_crop);
     }
 
+    // want to generate a pyramid but no resolution count specified
+    // calculate based upon the scale and image size
+    // the tile size has not been yet been set on the writer,
+    // and the writer needs a MetadataRetrieve with resolutions populated first
+    if (pyramidScale > 1 && pyramidResolutions == 1) {
+      pyramidResolutions = 0;
+      int checkWidth = width;
+      int checkHeight = height;
+      while (checkWidth > saveTileWidth && checkHeight > saveTileHeight) {
```

Comment:
With a command along the lines of `bfconvert -pyramid-scale 2 "test&sizeX=2048&sizeY=2048.fake" test2.ome.tiff -tilex 1024 -tiley 1024` this results in a single resolution.

For tiled based writing, the following should create 2 resolution levels in this scenario and align with the behavior of `bioformats2raw "test&sizeX=2048&sizeY=2048.fake" --target-min-size=1024 test.zarr`

```suggestion
      while (checkWidth >= saveTileWidth && checkHeight >= saveTileHeight) {
```

Maybe this should also check the height/width are greater than `pyramidScale` as well to avoid `0x0` image in the case of stripped images